dex_helper.cc

```diff
@@ -646,7 +646,7 @@ bool DexHelper::IsMethodMatch(size_t dex_id, uint32_t method_id, uint32_t return
     const auto &proto = dex.ProtoIds()[method.proto_idx];
     const auto &shorty = strs[proto.shorty_idx];
     if (return_type != uint32_t(-2) && proto.return_type_idx != return_type) return false;
-    if (!parameter_shorty.empty() && strcmp(parameter_shorty.data(), shorty.data() + 1) != 0) return false;
+    if (!parameter_shorty.empty() && parameter_shorty.compare(shorty.data() + 1) != 0) return false;
```

Comment:
still not c++ style

Reply:
what is c++ style, used substring?

Reply:
`string_view::data()` does NOT necessarily end with `\0`, so try not to use it.